### PR TITLE
implemented filter functionality for albums page

### PR DESCRIPTION
**Pull request by developingright**
Hi @razzeee,
I have implemented the filter functionality for the albums page. Please let me know if any changes are required.

filter2

### CODE REVIEW

File: src/SharedUtil.elm [hunk 1]
```diff
@@ -56,3 +63,144 @@ sortByRandom mySeed list =
     Random.step
         (Random.List.shuffle list)
         mySeed
+
+
+checkfilterbutton : List FilterButton -> List FilterButton
```
Comment (razzeee) [inline src/SharedUtil.elm:68]: these names should probably be using camelCase too?

File: src/SharedUtil.elm [hunk 2]
```diff
@@ -56,3 +63,144 @@ sortByRandom mySeed list =
...
+            if id == idx then
+                { task | name = task.name, state = not task.state }
+
+            else
+                { task | name = task.name, state = task.state }
+    in
+    List.indexedMap toggle list
+
+
+sort_filter_album : AlbumSort -> List AlbumObj -> List AlbumObj
```
Comment (razzeee) [inline src/SharedUtil.elm:86]: Why a snake case name?

**Comment by developingright** [comment]
Hi @razzeee,
I have camelCased the names of the function. I used snake case names for the functions earlier because in some places I thought it was more readable when written in that manner.

File: src/Pages/Music/Albums.elm [hunk 3]
```diff
@@ -268,3 +437,84 @@ currentButtonText currentButton name ( isCurrent, button ) =
 
         False ->
             row [ Font.color Colors.navText ] [ Element.text name ]
+
+
+filterfieldbutton : msg -> String -> List FilterButton -> Element msg
```
Comment (razzeee) [inline src/Pages/Music/Albums.elm:442]: These should be camelCase too

Comment (developingright) [reply src/Pages/Music/Albums.elm]: Done :+1:

**Pull request merged**